### PR TITLE
Adapt to the JerryScript v2.0

### PR DESCRIPTION
**Pull request by kkristof**
Signed-off-by: Kristof Kosztyo <kkristof@inf.u-szeged.hu>

### CODE REVIEW

**Comment by kkristof** [comment]
The API has been changed for the `jerry_get_object_native_pointer` in the JerryScript v2.0.

File: src/gepardSurfaceBinding.cpp [hunk 1]
```diff
@@ -130,29 +130,27 @@ void bindSurface()
...
 }
 
 gepard::Surface* getNativeSurfacePtr(jerry_value_t object)
 {
     void* nativePointer = nullptr;
-    const jerry_object_native_info_t *type_p = nullptr;
-    bool hasNativePointer = jerry_get_object_native_pointer(object, &nativePointer, &type_p);
-    if (hasNativePointer && (type_p == &native_x_surface_type_info || type_p == &native_memory_surface_type_info))
+    bool hasNativePointer = jerry_get_object_native_pointer(object, &nativePointer, &native_x_surface_type_info)
+            || jerry_get_object_native_pointer(object, &nativePointer, &native_memory_surface_type_info);
```
Comment (elecro) [inline src/gepardSurfaceBinding.cpp:152]: Do we need two different free callbacks for the surface types? Both of the free callbacks calls `delete` on the pointer (and there is also a bit of debug msg what object is freed). If I remember correctly the Surface has a virtual destructor, thus we can simply cast the pointer to `Surface*` and invoke the `delete` on it.

Comment (kkristof) [reply src/gepardSurfaceBinding.cpp:152]: Yes, we need. In some cases we need them to access the surface's extra functionality i.e. `hasToQuit`, however the Gepard internally doesn't care about the surface's type.

**Review by elecro — APPROVED** [review]
lgtm

**Comment by kkristof** [comment]
I'm closing this as I'm not sure if it is compatible with the current JerryScript API. I'm going to adapt and create a new merge request with the necessary changes.

**Pull request closed without merging**